### ISSUE TITLE
Adding third user to the Chat with Direct line

### ISSUE THREAD

**Issue by jrom20** [Microsoft/BotBuilder#2334]
Hi,

I was trying to generate a conversation which initially it will start with a bot and default user once this default user requests to talk with another user this will join this third user, and they will be able talk between them, right now I already achieved to read the conversation in the third user chat, bad when I try to send a message via Direct line in that specific conversation it is not shown in the initial user chat.

Do you know if there is any way to achieve this? I am sending this message in the following way using c#.

How can this third user get into the conversation?

```
DirectLineClient client = new DirectLineClient(directLineSecret);

var conversationId = "XXXXXXXXXXX";

new System.Threading.Thread(async () => await ReadBotMessagesAsync(client,conversationId)).Start();
            
Activity userMessage = new Activity
                        {
                            From = new ChannelAccount("User09090", "User User"),
                            Text = "Hello everyone!",
                            Type = ActivityTypes.Message
                    };

  await client.Conversations.PostActivityAsync(conversationId, userMessage);
```

Thank you in advance guys.

**Comment by dandriscoll** [comment]
Hi @jrom20, this should work, although it hasn't been tested very thoroughly -- we're focused on the single-user scenario at the moment.

Can you send an example conversation ID and I can take a look?

**Comment by jrom20** [comment]
@dandriscoll  I'm already able to join a third user that specific conversationId, and reply into it. but in the first user side I can't let him know that there is another user, if I were able to know into the first user chat side that there is third user I would change the replying  program. otherwise I noticed that the third user get into the current initial user dialog, there is any option to stop the dialog? I'm doing this right now, and it shows the following error, "Sorry, my bot code is having an issue."

Steps:
First I start a conversation between Bot and User1, and User1 enter into the default Dialog (Using LUIS),

`await Conversation.SendAsync(activity, () => new IniatialDialog());`

When a third user get into the conversation I close the current dialog or user state doing this:

`activity.GetStateClient().BotState.DeleteStateForUser(activity.ChannelId, activity.From.Id);`

I'm doing this because third user was getting into the default dialog, and I don't want that, so I created a new Dialog to save the new third user Id in ConversationData, and then I start a new dialog.

await Conversation.SendAsync(activity, () => new UserToUserDialog());

But It shows the below error that I mentioned.

Conversation Id FA92v5iMfuw3HMm0ekRsTp.
